File: app/src/main/java/com/example/mvx/mvx/android/guide/learn/project/annotation/SecondActivity.java
package com.example.mvx.mvx.android.guide.learn.project.annotation;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.example.mvx.mvx.android.guide.learn.project.R;

public class SecondActivity extends AppCompatActivity {

    @InjectIntent("age")
    private int age;

    @InjectIntent("married")
    private boolean married;

    @InjectIntent("sex")
    private String sex;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        try {
            InjectData.inject(this);
        } catch (IllegalAccessException e) {
            throw new RuntimeException(e);
        }
        initView();
    }

    private void initView() {
        TextView textView1 = findViewById(R.id.text1);
        textView1.setText(age + "");

        TextView textView2 = findViewById(R.id.text2);
        textView2.setText(sex);

        TextView textView3 = findViewById(R.id.text3);
        textView3.setText(married + "");
    }

}

File: app/src/main/java/com/example/mvx/mvx/android/guide/learn/project/annotation/SecondActivity.kt
package com.example.mvx.mvx.android.guide.learn.project.annotation

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.mvx.mvx.android.guide.learn.project.R
import com.example.mvx.mvx.android.guide.learn.project.annotation.InjectData.inject
import com.example.mvx.mvx.android.guide.learn.project.databinding.ActivitySecondBinding

class SecondActivity : AppCompatActivity() {
    @InjectIntent("age")
    private val age = 0

    @InjectIntent("married")
    private val married = false

    @InjectIntent("sex")
    private val sex: String? = null

    private lateinit var binding: ActivitySecondBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySecondBinding.inflate(layoutInflater)
        setContentView(binding.root)
        try {
            inject(this)
        } catch (e: IllegalAccessException) {
            throw RuntimeException(e)
        }
        initView()
    }

    private fun initView() {
        binding.text1.text = age.toString() + ""
        binding.text2.text = sex
        binding.text3.text = married.toString() + ""
    }
}